integration-tests/src/test/java/gov/uk/di/ipv/cri/common/api/stepDefinitions/APISteps.java

```diff
@@ -13,25 +13,26 @@
 import java.net.URISyntaxException;
 import java.net.http.HttpResponse;
 import java.util.Map;
-import java.util.Optional;
 import java.util.UUID;
+import java.util.logging.Logger;
 
 import static gov.uk.di.ipv.cri.common.api.util.IpvCoreStubUtil.sendCreateAuthCodeRequest;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 
 public class APISteps {
-
+    private static final Logger LOG = Logger.getLogger(APISteps.class.getName());
     private static final String ENVIRONMENT = "/dev"; // dev, build, staging, integration
     private static String devSessionUri;
     private static String devAuthorizationUri;
     public static String devAccessTokenUri;
     private static final ObjectMapper objectMapper = new ObjectMapper();
+    private static final String REDIRECT_URI = System.getenv("IPV_CORE_STUB_URL");
     private static final String DEFAULT_REDIRECT_URI =
-            Optional.ofNullable(System.getenv("IPV_CORE_STUB_URL") + "/callback")
-                    .orElse("https://di-ipv-core-stub.london.cloudapps.digital/callback");
-    private static final String DEFAULT_CLIENT_ID =
-            Optional.ofNullable(System.getenv("DEFAULT_CLIENT_ID")).orElse("ipv-core-stub");
+            (REDIRECT_URI.toLowerCase().startsWith("http"))
+                    ? REDIRECT_URI + "/callback"
+                    : "https://di-ipv-core-stub.london.cloudapps.digital/callback";
+    private static final String DEFAULT_CLIENT_ID = System.getenv("DEFAULT_CLIENT_ID");
```

Comment:
Please could we put this back to `Optional.ofNullable(System.getenv("DEFAULT_CLIENT_ID")).orElse("ipv-core-stub");`

Reply:
Done